lib/main.dart

```diff
@@ -113,24 +113,26 @@ Future<void> main({AppConfig appConfig = schulCloudAppConfig}) async {
     await services.allReady();
 
     // Set demo banner based on current user.
-    // TODO(marcelgarus): dispose id stream. / use switchMap
+    StreamAndData<User, CachedFetchStreamData<dynamic>> userStream;
     services.storage.userIdString
         .map((idString) => Id<User>(idString))
         .listen((userId) {
-      userId.resolve().listen((user) {
-        // TODO(marcelgarus): Don't hardcode role id.
-        final isDemo = [
-          Id<Role>('0000d186816abba584714d00'), // demo general
-          Id<Role>('0000d186816abba584714d02'), // demo student
-          Id<Role>('0000d186816abba584714d03'), // demo teacher
-        ].any((demoRole) => user?.roleIds?.contains(demoRole) ?? false);
+      userStream?.dispose();
+      userStream = userId.resolve()
+        ..listen((user) {
```

Comment:
Can we just use `Stream.switchMap()` from <kbd>rxdart</kbd> here?

Reply:
I don't think so because afaik it doesn't allow us to dispose the previous stream when a new one arrives…